cmd/restic/cmd_mount.go

```diff
@@ -79,9 +81,38 @@ func init() {
 	mountFlags.StringArrayVar(&mountOptions.Paths, "path", nil, "only consider snapshots which include this (absolute) `path`")
 
 	mountFlags.StringVar(&mountOptions.SnapshotTemplate, "snapshot-template", time.RFC3339, "set `template` to use for snapshot dirs")
+	mountFlags.BoolVar(&mountOptions.DaemonMode, "daemon", false, "run this mount in background like a daemon")
+}
+
+func linearSearch(args []string, search string) int {
+	for i, n := range args {
+		if n == search {
+			return i
+		}
+	}
+	return -1
+}
+
+func removeElement(args []string, search string) []string {
+	index := linearSearch(args, search)
+	if index != -1 {
+		return append(args[:index], args[index+1:]...)
+	}
+	return args
 }
 
 func runMount(opts MountOptions, gopts GlobalOptions, args []string) error {
+	if opts.DaemonMode {
+		filename := os.Args[0]
+		attr := new(syscall.ProcAttr)
```

Comment:
A daemon should start a new session and chdir to the root:

```go
attr := &syscall.ProcAttr{
	Dir: "/",
	Sys: &syscall.SysProcAttr{Setsid: true},
}
```

Maybe Noccty should be set as well?